storage/cloud-client/hmac_test.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for hmac.py. Requires GOOGLE_CLOUD_PROJECT (valid project) and
+HMAC_KEY_TEST_SERVICE_ACCOUNT (valid service account email) env variables to be
+set in order to run.
+"""
+
+
+import os
+
+from google.cloud import storage
+import pytest
+
+import hmac
+
+
+PROJECT_ID = os.environ['GOOGLE_CLOUD_PROJECT']
+SERVICE_ACCOUNT_EMAIL = os.environ['HMAC_KEY_TEST_SERVICE_ACCOUNT']
+
+
+@pytest.fixture()
+def new_hmac_key():
+    """
+    Fixture to create a new HMAC key, and to guarantee all keys are deleted at
+    the end of each test.
+    """
+    hmac_key = hmac.create_key(PROJECT_ID, SERVICE_ACCOUNT_EMAIL)
+    yield hmac_key
+    storage_client = storage.Client(project=PROJECT_ID)
+    hmac_key = storage_client.get_hmac_key_metadata(hmac_key.access_id,
+                                                    project_id=PROJECT_ID)
+    if hmac_key.state == 'DELETED':
+        return
+    if not hmac_key.state == 'INACTIVE':
```

Comment:
Can we not delete the key if it isn't `INACTIVE`? If so, should the delete snippet clarify that and perform the same check?

Reply:
Yes, only inactive keys can be deleted. The delete snippet does mention that the access_id should be for an inactive hmac key (I basically just copied what was in the canonical sample, https://github.com/googleapis/nodejs-storage/blob/master/samples/hmacKeyDelete.js ).

Reply:
I think adding a comment before the deletion (in the snippet) might be more noticeable for users (I missed it), but up to you.

Reply:
I edited the comment to make it hopefully a bit more obvious.